src/PKSim.Core/Services/ParameterContainerTask.cs

```diff
@@ -51,30 +56,38 @@ public ParameterContainerTask(IParameterQuery parameterQuery, IParameterFactory
       {
-         addParametersTo(parameterContainer, originData, originData.AllCalculationMethods().Select(cm => cm.Name), 
+         addParametersTo(parameterContainer, originData, originData.AllCalculationMethods().Select(cm => cm.Name),
             param => param.BuildingBlockType == PKSimBuildingBlockType.Individual && string.Equals(param.ParameterName, parameterName));
       }
 
-      public void AddDiseaseStateParametersTo(DiseaseState diseaseState)
-      {
-         addParametersTo(diseaseState, null, CoreConstants.CalculationMethod.ForDiseaseStates);
-      }
+      public void AddDiseaseStateParametersTo(DiseaseState diseaseState) => addParametersTo(diseaseState, calculationMethods: CoreConstants.CalculationMethod.ForDiseaseStates);
 
-      public void AddModelParametersTo<TContainer>(TContainer parameterContainer, OriginData originData, ModelProperties modelProperties, IFormulaCache formulaCache) where TContainer : IContainer
+      public void AddParametersToSpatialStructureContainer<TContainer>(TContainer parameterContainer, OriginData originData, ModelProperties modelProperties, IFormulaCache formulaCache) where TContainer : IContainer
       {
-         bool addParameter(ParameterMetaData param)
+         bool addParameter(ParameterMetaData parameterMetaData)
          {
-            return param.BuildingBlockType != PKSimBuildingBlockType.Individual || _commonParametersRepository.UsedForAllSpecies(param);
+            //all non individual parameters are added to the spatial structure by default
+            //otherwise, we add all parameters that are used by all species. The value used will be set based on the fact that the parameter has a shared value for all parameters or species or not
+            return parameterMetaData.BuildingBlockType != PKSimBuildingBlockType.Individual || _individualParameterBySpeciesRepository.UsedForAllSpecies(parameterMetaData);
          }
 
-         addParametersTo(parameterContainer, originData, modelProperties.AllCalculationMethods().Select(cm => cm.Name), addParameter, formulaCache);
-      }
 
-      public void AddActiveProcessParametersTo(CompoundProcess process)
-      {
-         addParametersTo(process, null, CoreConstants.CalculationMethod.ForProcesses);
+         void parameterValueModifier(ParameterMetaData parameterMetaData, IParameter parameter)
+         {
+            //here we know that the parameter fulfills the criteria above. However, in case of a parameter that is used by all species, we need to verify that either the formula or the value is common 
+            //to all species. Otherwise, we set the value to NaN
+            var (isSame, exists) = _sameFormulaOrValueForAllSpeciesRepository.IsSameFormulaOrValue(parameterMetaData);
+            if (exists && !isSame)
+            {
+               parameter.Formula = new ConstantFormula(double.NaN);
```

Comment:
I believe isSameFormulaOrValue returns TRUE for distributed parmaeter? Is this possible? I need to check

Reply:
same